.gitignore

```diff
@@ -20,3 +20,7 @@ clangs
 bin/test
 bin/unittest
 .DS_Store
+*.so.*
+*.so
+/bin/*
+*~
```

Comment:
Please remove that. It should not be part of project specific ignore files. If you want to ignore that I recommend either adding a global ignore file to ignore that across all git projects or add it to the private ignore file for the project. You should be able to find this by searching online.